osbs/build/build_request.py

```diff
@@ -1060,11 +1060,25 @@ def render_tag_from_config(self):
         unique_tag = self.spec.image_tag.value.split(':')[-1]
         tag_suffixes = {'unique': [unique_tag], 'primary': []}
 
+        # this part is very ugly but otherwise ~900 tests will fail
+        # as self.repo_info object is not injected properly
+        tags = []
+        if self._repo_info:
+            tags = self._repo_info.configuration.container.get('tags', [])
+
         if self.spec.build_type.value == BUILD_TYPE_ORCHESTRATOR:
             if self.scratch:
                 pass
             elif self.isolated:
                 tag_suffixes['primary'].extend(['{version}-{release}'])
+            elif tags:
+                tag_suffixes['primary'].extend(['{version}-{release}'])
+                for tag in tags:
+                    assert '-' not in tag, \
```

Comment:
Validation should extend beyond this. `AdditionalTagsConfig` uses a regex to do so. Having this validation in two distinct places is error prone.